dartagnan/src/main/java/com/dat3m/dartagnan/encoding/EncodingContext.java

```diff
@@ -219,7 +219,7 @@ public Formula lastValue(MemoryObject base, int offset) {
         //TODO match this with the actual type stored at the memory address
         // (we do not know and guess the arch type right now)
         TypeFactory types = TypeFactory.getInstance();
-        int archSize = types.getMemorySizeInBytes(types.getArchType());
+        int archSize = types.getMemorySizeInBytes(types.getArchType()) * 8;
```

Comment:
is this pattern used somewhere else? Would it make sense to have a method `getMemorySize()` in the type factory which encapsulates the multiplication??

Reply:
No, it is never used. But again, this is more or less a randomly chosen value. Just using the constant `64` might be even better than relying on `archType`.